packages/react-static/src/index.d.ts

```diff
@@ -584,6 +584,22 @@ export interface PathsConfig {
 
 export type DocumentComponent<T extends object = any> = React.ComponentType<DocumentProps<T>>
 
+export interface InlineScript {
+  /**
+   * The script as a string.
+   */
+  script: string
+  /**
+   * The script's sha256 hash in base64 using the subresource integrity format,
+   * eg.: 'sha256-<base64-value>'. This value can be directly used in a CSP.
+   */
+  hash: string
+}
+
+export interface InlineScripts {
```

Comment:
We probably want "generic object, including this one":

```typescript
export interface InlineScripts extends Record<string, InlineScript> {
   routeInfo: InlineScript
}
```

This _should_ allow us to do this:

```typescript
const scripts: InlineScripts = {
  routeInfo: { script: '...', hash: '...' }
}
// this should work

scripts['anything'] = { script: '...', hash: '...' }
// this should work

scripts.
// this should autocomplete `routeInfo`
```

It then _disallows_:

```typescript
const scripts: InlineScripts = {
  routeInfo: { script: '...', hash: '...' }
}
scripts['foo'] = 'bar'
// Type '"bar"' is not assignable to type InlineScript
```

Reply:
Nice, added in https://github.com/react-static/react-static/pull/1373/commits/d76638ea10606be4e10d110883f449f9865f970c